src/twisted/runner/procmon.py

```diff
@@ -344,15 +344,18 @@ def startProcess(self, name):
         proto.name = name
         self.protocols[name] = proto
         self.timeStarted[name] = self._reactor.seconds()
-        self._reactor.spawnProcess(
-            proto,
-            process.args[0],
-            process.args,
-            uid=process.uid,
-            gid=process.gid,
-            env=process.env,
-            path=process.cwd,
-        )
+        try:
+            self._reactor.spawnProcess(
+                proto,
+                process.args[0],
+                process.args,
+                uid=process.uid,
+                gid=process.gid,
+                env=process.env,
+                path=process.cwd,
+            )
+        except OSError:
```

Comment:
Is is ok to discard the reason / details of this error?

Reply:
Also, do we want to retry on any OSError or only just on "temporary" errors like `Resource temporarily unavailable` ?

Should we retry if we get a permission denied error or a too many open files errors ?

Reply:
Also, is this even being handled in the right place? `IReactorProcess.spawnProcess` does not document a `@raises OSError`, which suggests that perhaps we ought to be delivering this error to `processExited` rather than raising it synchronously?

It's been raising `OSError` for a long time of course, maybe this is just a documentation issue that needs to be addressed, but I bet that this is, like, `WindowsError` sometimes instead